### ISSUE TITLE
Add '(De)Select all' buttons to Clean-up Configuration dialog and ability to search for a specific clean-up

### ISSUE THREAD

**Issue by HannesWell** [eclipse-jdt/eclipse.jdt.ui#119]
It would be handy if the Java  Code Style Clean-Up Preference dialog would have Buttons to `Deselect all` and `Select all` available clean ups. Especially the former avoids a lot of manual unchecking if one you want to have a clean configuration to then select only one or a few certain clean ups.

Furthermore it would super convenient if the Clean-Up preferences dialog would offer something like a search bar, where you can type in the name (label, id, etc.) and get the corresponding clean up elements displayed.
The Java -> Compiler Errors/Warnings dialog offers such search bar, but also has a different structure. Therefore I don't know how this should work for different tabs.

I assume that the dialog itself is the same for 
- `Preferences -> Java Code Style -> Clean Up -> Edit`
- `Preferences -> Java -> Editor -> Save Actions -> Configure (Additional actions)`
- `Right click on a project in the Explorer -> Source -> Clean up... -> Configure (Use custom profile)`

**Comment by vogella** [comment]
IIRC @jjohnstn was planning to look at this in the past but this might have been lost over time.

**Comment by jjohnstn** [comment]
The search option would require a complete rewrite of the page logic so I would say no.  I have though implemented four new buttons on the individual pages: Select All, Deselect All, Defaults, and Reset.  The Defaults button sets the options to their default values and the Reset button resets the values to their initial settings in case you want to start again.  The buttons apply to the options for the page shown.  I did not want to extend it to all options/all pages as the user can't even see what options are being changed and then there isn't any granularity.  For example, I could easily see a user setting all Performance options and/or Java Feature options but not wanting to set Source Changing options which have warnings about their usage.  Try it out and see if you like it.

**Comment by jjohnstn** [comment]
Hi @vogella Do you want to try the PR mentioned above as I believe you originally requested this via bugzilla?

**Comment by vogella** [comment]
> Hi @vogella Do you want to try the PR mentioned above as I believe you originally requested this via bugzilla?

Currently on vacation, so please merge. I can give feedback in 2 weeks

**Comment by jjohnstn** [comment]
Ok, I will wait until Monday and give Hannes a chance to comment.  I want to get it into M2 so there is time to respond to comments in M3.

**Comment by noopur2507** [comment]
The problems discussed in the reported issue are very similar to the ones that were present for Java > Formatter profile. For these reasons, the formatter page was rewritten to add search and modify all options. See [bug 462940](https://bugs.eclipse.org/bugs/show_bug.cgi?id=462940).

Adding so many buttons will surely clutter the UI. Also, the meaning and difference between "Defaults" and "Reset" is not clear with the button names. 

Do you think we need "Defaults" and "Reset" for each tab? Instead, we can add "Restore Defaults" and "Reset Profile" as common profile-level buttons before the "Apply" button. That looks more consistent with the rest of the Eclipse UI.

For the tab-level buttons, we can just keep the Select and Deselect All buttons.

Another option is to just add the two profile-level buttons. And move all four tab-level buttons to context menu options on each tab.

**Comment by jjohnstn** [comment]
Ok, will remove the Defaults and Reset buttons from the tab options and put them in the main dialog as common buttons.  As well, I will rename as specified.  I think context menu options will not be discovered by the end-user since there aren't any other options there previously.  I will update the patch.  Hannes was requesting the Deselect All for common profile to clear all to start.  If you agree, I can add it easily enough as I already did a patch with it before I saw your comment.
